pants_plugins/sendwave/pants_docker/docker_component.py

```diff
@@ -18,20 +16,5 @@ class DockerComponent:
 
 
 @union
-class DockerComponentRequest(ABC):
-    field_set_type: ClassVar[Type[FieldSet]]
-    fs: FieldSet
-
-    def __init__(self, field_set) -> None:
-        self.fs = field_set
-
-
-def from_dependencies(
-    tgts: Targets, um: UnionMembership
-) -> List[DockerComponentRequest]:
-    return [
-        request_type(request_type.field_set_type.create(tgt))
-        for tgt in tgts
-        for request_type in um[DockerComponentRequest]
-        if request_type.field_set_type.is_applicable(tgt)
-    ]
+class DockerComponentFieldSet:
+    pass
```

Comment:
Another way to do this is to have this subclass `FieldSet` and `metaclass=ABCMeta`, then have all "union members" subclass this. It's not at all necessary, but possibly is clearer.